### PR TITLE
qv2ray: Init at 2.6.3

### PR DESCRIPTION
**Pull request by poscat0x04**
###### Motivation for this change

qv2ray is a GUI frontend to the network proxying tool v2ray.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/qv2ray/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    clang
+    qt5.qmake 
+    qt5.qttools
+    protobuf
+    grpc
+    pkgconfig
+    c-ares
+    abseil-cpp
+  ];
+}
```
Comment (symphorien) [inline pkgs/applications/networking/qv2ray/default.nix:54]: the `meta` attribute is missing

File: pkgs/applications/networking/qv2ray/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  ];
+
+  preConfigure = ''
+    export _QV2RAY_BUILD_INFO_="Qv2ray Nixpkgs"
+    export _QV2RAY_BUILD_EXTRA_INFO_="(Nixpkgs build) $(uname -a | cut -d " " -f3,13)"
+  '';
+
+  installPhase = ''
+    cmake --install .
+  '';
```
Comment (symphorien) [inline pkgs/applications/networking/qv2ray/default.nix:39]: are you sure this is not the default?

File: pkgs/applications/networking/qv2ray/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DQV2RAY_DISABLE_AUTO_UPDATE=on"
+    "-DQV2RAY_TRANSLATION_PATH=$(out)/share/qv2ray/lang"
+  ];
+
+  preConfigure = ''
+    export _QV2RAY_BUILD_INFO_="Qv2ray Nixpkgs"
+    export _QV2RAY_BUILD_EXTRA_INFO_="(Nixpkgs build) $(uname -a | cut -d " " -f3,13)"
```
Comment (symphorien) [inline pkgs/applications/networking/qv2ray/default.nix:34]: including the uname is impure. Better use something generic like "nixpkgs".

File: pkgs/applications/networking/qv2ray/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    repo = "Qv2ray";
+    rev = "v${version}";
+    sha256 = "12x72rsd5iaj06qmma8x6g7syqaiixnfb37f3ppx8x00c77s78va";
+    fetchSubmodules = true;
+  };
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DQV2RAY_DISABLE_AUTO_UPDATE=on"
+    "-DQV2RAY_TRANSLATION_PATH=$(out)/share/qv2ray/lang"
```
Comment (symphorien) [inline pkgs/applications/networking/qv2ray/default.nix:29]:
```suggestion
    "-DQV2RAY_TRANSLATION_PATH=${placeholder "out"}/share/qv2ray/lang"
```

File: pkgs/applications/networking/qv2ray/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  nativeBuildInputs = [ 
+    cmake
+    clang
+    qt5.qmake 
+    qt5.qttools
+    protobuf
+    grpc
+    pkgconfig
+    c-ares
+    abseil-cpp
```
Comment (symphorien) [inline pkgs/applications/networking/qv2ray/default.nix:52]: libraries like abseil-cpp (those which contain code that is embedded in the resulting binaries) are supposed to go to `buildInputs`.
(as opposed to build tools which are executed during the build, like cmake, which you correctly placed in `nativeBuildInputs`. )

If unsure, you can try to build with `strictDeps  = true`, and it should fail to build if you made a mistake.

File: pkgs/applications/networking/qv2ray/default.nix (hunk 2 above)
Comment (poscat0x04) [reply pkgs/applications/networking/qv2ray/default.nix:39]: Yep that seems to be the default. I was not sure cuz I tried to grep "cmake --install" and didn't find anything.

File: pkgs/applications/networking/qv2ray/default.nix [hunk 6]
```diff
@@ -0,0 +1,54 @@
+{ stdenv
+, fetchFromGitHub
+, qt5
```
Comment (symphorien) [inline pkgs/applications/networking/qv2ray/default.nix:3]: you should not have `qt5` as argument but use `libsForQt5.callPackage` instead. See https://nixos.org/nixpkgs/manual/#sec-language-qt

File: pkgs/applications/networking/qv2ray/default.nix [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    export _QV2RAY_BUILD_EXTRA_INFO_="(Nixpkgs build) $(uname -a | cut -d " " -f3,13)"
+  '';
+
+  installPhase = ''
+    cmake --install .
+  '';
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [
```
Comment (symphorien) [inline pkgs/applications/networking/qv2ray/default.nix:43]: I expect this is lacking `wrapQtAppsHook`.

**Comment by poscat0x04** [comment]
Thank you for your generous help! I'm terribly sorry that I forgot to read the nixpkgs manual before opening this PR and I'll try to be less careless next time.

File: pkgs/applications/networking/qv2ray/default.nix [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    clang
+    pkgconfig
+    qmake
+    qttools
+    wrapQtAppsHook
+  ];
+
+  meta = with stdenv.lib; {
+    description = "An GUI frontend to v2ray";
+    homepage = https://qv2ray.github.io/en/;
```
Comment (symphorien) [inline pkgs/applications/networking/qv2ray/default.nix:64]:
```suggestion
    homepage = "https://qv2ray.github.io/en/";
```
due to https://github.com/NixOS/rfcs/pull/45/files

**Review by symphorien — APPROVED** [review]
compiles fine with nix-review and a window shows up when launched.

File: pkgs/applications/networking/qv2ray/default.nix [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    "-DQV2RAY_DISABLE_AUTO_UPDATE=on"
+    "-DQV2RAY_TRANSLATION_PATH=${placeholder "out"}/share/qv2ray/lang"
+  ];
+
+  preConfigure = ''
+    export _QV2RAY_BUILD_INFO_="Qv2ray Nixpkgs"
+    export _QV2RAY_BUILD_EXTRA_INFO_="(Nixpkgs build) nixpkgs"
+  '';
+
+  enableParallelBuilding = true;
```
Comment (IvarWithoutBones) [inline pkgs/applications/networking/qv2ray/default.nix:41]: This is not required, nix figures this out automatically. As the nixpkgs manual states:
`Unless set to false, some build systems with good support for parallel building including cmake, meson, and qmake will set it to true.`

File: maintainers/maintainer-list.nix [hunk 10]
```diff
@@ -6178,6 +6178,18 @@
     githubId = 138074;
     name = "Pedro Pombeiro";
   };
+  poscat = {
```
Comment (IvarWithoutBones) [inline maintainers/maintainer-list.nix:6181]: This should be in a seperate commit. Speaking of commits, the others should be squashed

File: pkgs/applications/networking/qv2ray/default.nix (hunk 8 above)
Comment (jonringer) [reply pkgs/applications/networking/qv2ray/default.nix:62]: this is still relevant

File: pkgs/applications/networking/qv2ray/default.nix [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    wrapQtAppsHook
+  ];
+
+  meta = with stdenv.lib; {
+    description = "An GUI frontend to v2ray";
+    homepage = https://qv2ray.github.io/en/;
+    license = licenses.gpl3;
+    maintainers = with maintainers; [
+      poscat
+    ];
```
Comment (jonringer) [inline pkgs/applications/networking/qv2ray/default.nix:66]:
```suggestion
    maintainers = with maintainers; [ poscat ];
```

File: pkgs/applications/networking/qv2ray/default.nix [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+    c-ares
+  ];
+
+  nativeBuildInputs = [ 
+    cmake
+    clang
+    pkgconfig
+    qmake
+    qttools
+    wrapQtAppsHook
```
Comment (jonringer) [inline pkgs/applications/networking/qv2ray/default.nix:57]:
```suggestion
```

File: pkgs/applications/networking/qv2ray/default.nix [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+, protobuf
+, openssl
+, pkgconfig
+, c-ares
+, abseil-cpp
+, libGL
+, zlib
+}:
+
+stdenv.mkDerivation rec {
```
Comment (jonringer) [inline pkgs/applications/networking/qv2ray/default.nix:18]:
```suggestion
mkDerivation rec {
```

File: pkgs/applications/networking/qv2ray/default.nix [hunk 14]
```diff
@@ -0,0 +1,69 @@
+{ stdenv
```
Comment (jonringer) [inline pkgs/applications/networking/qv2ray/default.nix:1]:
```suggestion
{ stdenv
, mkDerivation
```

**Review by jonringer — COMMENTED** [review]
use qt5's mkDerivation, it will automatically wrap the programs for you

**Comment by poscat0x04** [comment]
ping @jonringer

**Review by marsam — APPROVED** [review]

**Pull request merged**